map.go

```diff
@@ -508,42 +527,55 @@ func (m *Map) Lookup(key, valueOut interface{}) error {
 //
 // Returns ErrKeyNotExist if the key doesn't exist.
 func (m *Map) LookupAndDelete(key, valueOut interface{}) error {
-	valuePtr, valueBytes := makeBuffer(valueOut, m.fullValueSize)
+	return m.lookupAndDelete(key, valueOut, 0)
+}
 
-	keyPtr, err := m.marshalKey(key)
-	if err != nil {
-		return fmt.Errorf("can't marshal key: %w", err)
-	}
+// LookupAndDeleteWithFlags retrieves and deletes a value from a Map.
+//
+// Passing FLock flag will look up and delete the value of a spin-locked
+// map without returning the lock. This must be specified if the elements
+// contain a spinlock.
+//
+// Returns ErrKeyNotExist if the key doesn't exist.
+func (m *Map) LookupAndDeleteWithFlags(key, valueOut interface{}, flags MapUpdateFlags) error {
+	return m.lookupAndDelete(key, valueOut, flags)
+}
 
-	attr := sys.MapLookupAndDeleteElemAttr{
-		MapFd: m.fd.Uint(),
-		Key:   keyPtr,
-		Value: valuePtr,
-	}
+// LookupBytes gets a value from Map.
+//
+// Returns a nil value if a key doesn't exist.
+func (m *Map) LookupBytes(key interface{}) ([]byte, error) {
+	valueBytes := make([]byte, m.fullValueSize)
+	valuePtr := sys.NewSlicePointer(valueBytes)
 
-	if err := sys.MapLookupAndDeleteElem(&attr); err != nil {
-		return fmt.Errorf("lookup and delete: %w", wrapMapError(err))
+	err := m.lookup(key, valuePtr, 0)
+	if errors.Is(err, ErrKeyNotExist) {
+		return nil, nil
 	}
 
-	return m.unmarshalValue(valueOut, valueBytes)
+	return valueBytes, err
 }
 
-// LookupBytes gets a value from Map.
+// LookupBytesWithFlags gets a value from Map.
+//
+// Passing FLock flag will look up the value of a spin-locked
+// map without returning the lock. This must be specified if the
+// elements contain a spinlock.
 //
 // Returns a nil value if a key doesn't exist.
-func (m *Map) LookupBytes(key interface{}) ([]byte, error) {
+func (m *Map) LookupBytesWithFlags(key interface{}, flags MapUpdateFlags) ([]byte, error) {
```

Comment:
Skip this, it's possible to do this via `LookupWithFlags`

Reply:
Addressed in c379cbfad81f51a1bc3eda17cbda2f674abe1c92